custom_components/tuya_local/devices/dreamegg_nite_1.yaml

```diff
@@ -0,0 +1,247 @@
+
+  - entity: number
+    name: Volume
+    icon: "mdi:volume-high"
+    dps:
+      - id: 9
+        name: value
+        type: integer
+        unit: "%"
+        range:
+          min: 10
+          max: 100
+        mapping:
+          - scale: 100
+
+  - entity: select
+    name: Sounds
+    icon: "mdi:waveform"
+    dps:
+      - id: 10
+        name: option
+        type: string
+        mapping:
+          - dps_val: "1"
+            value: Bird
+          - dps_val: "2"
+            value: Cricket
+          - dps_val: "3"
+            value: Campfire
+          - dps_val: "4"
+            value: Gentle Rain
+          - dps_val: "5"
+            value: Heavy Rain
+          - dps_val: "6"
+            value: Rainstorm
+          - dps_val: "7"
+            value: Brook
+          - dps_val: "8"
+            value: Gentle Surf
+          - dps_val: "9"
+            value: Sea Wave
+          - dps_val: "10"
+            value: Waterfall
+          - dps_val: "11"
+            value: Wind
+          - dps_val: "12"
+            value: Sushing
+          - dps_val: "13"
+            value: Womb Sound
+          - dps_val: "14"
+            value: Hair Dryer
+          - dps_val: "15"
+            value: Airplane
+          - dps_val: "16"
+            value: Cruise Ship
+          - dps_val: "17"
+            value: Train
+          - dps_val: "18"
+            value: Wind Chimes
+          - dps_val: "19"
+            value: White Noise
+          - dps_val: "20"
+            value: Pink Noise
+          - dps_val: "21"
+            value: Brown Noise
+          - dps_val: "22"
+            value: Gentle Fan
+          - dps_val: "23"
+            value: Box Fan
+          - dps_val: "24"
+            value: Oscillating Fan
+          - dps_val: "25"
+            value: Zen
+          - dps_val: "26"
+            value: Meditation
+          - dps_val: "27"
+            value: Lullaby
+          - dps_val: "28"
+            value: Little Star
+          - dps_val: "29"
+            value: Bedtime
+          - dps_val: "30"
+            value: Brahm's Lullaby
+          - dps_val: "31"
+            value: Music Box
+          - dps_val: "32"
+            value: Piano
+          - dps_val: "33"
+            value: Mozart
+          - dps_val: "34"
+            value: Morning
+
+  - entity: select
+    name: Favorites
+    icon: "mdi:palette"
+    dps:
+      - id: 13
+        type: string
+        name: option
+        optional: true
+        mapping:
+          - dps_val: mode_1
+            value: Scene 1
+          - dps_val: mode_2
+            value: Scene 2
+          - dps_val: mode_3
+            value: Scene 3
+          - dps_val: mode_4
+            value: Scene 4
+          - dps_val: mode_5
+            value: Scene 5
+          - dps_val: mode_6
+            value: Scene 6
+          - dps_val: mode_7
+            value: Scene 7
+          - dps_val: mode_8
+            value: Scene 8
+
+  - entity: number
```

Comment:
Would be better as a time entity (both time and number have issues - time only works properly if your HA settings are for 24h time, but number is not user friendly when it goes up to 24h in minutes (or seconds)

Reply:
Tried time entity, however, the HA UI seems to indicate it's a clock alarm (e.g. to go off at a specific time rather than a duration of time) so I'll leave it as a number entity since it's for a time duration.